---
citation-template: v0.2.0
creation-date: 2025:01:24-15:22:53
---

%% Note Body --------------------------------------------------- %%
# Critical assessment of E. Coli genome-scale metabolic model with high-throughput mutant fitness data

### Meta
- ** citekey **: bernsteinCriticalAssessmentColi2023
- ** authors **: David B. Bernstein, Batu Akkas, Morgan N. Price, Adam P. Arkin
- ** year **: [[2023]]
- ** doi **: https://doi.org/10.1101/2023.01.05.522875
- ** publication **: 
- ** Web ** : [Open online](https://www.biorxiv.org/content/10.1101/2023.01.05.522875v1)


### Abstract:
The E. coli genome-scale metabolic model (GEM) is a gold standard for the simulation of cellular metabolism. Experimental validation of model predictions is essential to pinpoint model uncertainty and ensure continued development of accurate models. Here we assessed the accuracy of the E. coli GEM using published mutant fitness data for the growth of gene knockout mutants across thousands of genes and 25 different carbon sources. We explored the progress of the E. coli GEM versions over time and further investigated errors in the latest version of the model (iML1515). We observed that model size is increasing while prediction accuracy is decreasing. We identified several adjustments that improve model accuracy â€“ the addition of vitamins/cofactors and re-assignment of reaction reversibility and isoenzyme gene to reaction mapping. Furthermore, we applied a machine learning approach which identified hydrogen ion exchange and central metabolism branch points as important determinants of model accuracy. Continued integration of experimental data to validate GEMs will improve predictive modeling of the mapping from genotype to metabolic phenotype in E. coli and beyond. <img class="highwire-fragment fragment-image" alt="Figure" src="https://www.biorxiv.org/content/biorxiv/early/2023/01/05/2023.01.05.522875/F1.medium.gif" width="440" height="144"/>E. coli genome-scale metabolic model flux balance analysis (FBA) prediction accuracy was quantified with published experimental data assaying gene knockout mutant growth across different carbon sources. Insights into model development trends and sources of inaccuracy were revealed. Model representational power (size) has been increasing over time, while accuracy has been decreasing.Adding vitamins/cofactors to the model environment and re-assigning reaction reversibility and isoenzyme gene-to-reaction mapping improves correspondence between model predictions and experimental data.Machine learning reveals hydrogen ion exchange and central metabolism branch points as important features in the determination of model accuracy.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead