---
citation-template: v0.2.0
creation-date: 2024:12:04-18:10:53
---

%% Note Body --------------------------------------------------- %%
# Metabolic network percolation quantifies biosynthetic capabilities across the human oral microbiome

### Meta
- ** citekey **: bernsteinMetabolicNetworkPercolation2019
- ** authors **: David B Bernstein, Floyd E Dewhirst, [[Daniel Segr√®]]
- ** year **: [[2019]]
- ** doi **: https://doi.org/10.7554/eLife.39733
- ** publication **: eLife
- ** Web ** : [Open online](https://elifesciences.org/articles/39733)


### Abstract:
The biosynthetic capabilities of microbes underlie their growth and interactions, playing a prominent role in microbial community structure. For large, diverse microbial communities, prediction of these capabilities is limited by uncertainty about metabolic functions and environmental conditions. To address this challenge, we propose a probabilistic method, inspired by percolation theory, to computationally quantify how robustly a genome-derived metabolic network produces a given set of metabolites under an ensemble of variable environments. We used this method to compile an atlas of predicted biosynthetic capabilities for 97 metabolites across 456 human oral microbes. This atlas captures taxonomically-related trends in biomass composition, and makes it possible to estimate inter-microbial metabolic distances that correlate with microbial co-occurrences. We also found a distinct cluster of fastidious/uncultivated taxa, including several Saccharibacteria (TM7) species, characterized by their abundant metabolic deficiencies. By embracing uncertainty, our approach can be broadly applied to understanding metabolic interactions in complex microbial ecosystems.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead
#Project/2024/RegulationImageMC 