---
citation-template: v0.2.0
creation-date: #!cdate
---

%% Note Body --------------------------------------------------- %%
# The E. Coli molecular phenotype under different growth conditions

### Meta
- ** citekey **: caglarColiMolecularPhenotype2017
- ** authors **: Mehmet U. Caglar, John R. Houser, Craig S. Barnhart, Daniel R. Boutz, Sean M. Carroll, Aurko Dasgupta, Walter F. Lenoir, Bartram L. Smith, Viswanadham Sridhara, Dariya K. Sydykova, Drew Vander Wood, Christopher J. Marx, Edward M. Marcotte, Jeffrey E. Barrick, Claus O. Wilke
- ** year **: [[2017]]
- ** doi **: https://doi.org/10.1038/srep45303
- ** publication **: Scientific Reports
- ** Web ** : [Open online](https://www.nature.com/articles/srep45303)


### Abstract:
Abstract Modern systems biology requires extensive, carefully curated measurements of cellular components in response to different environmental conditions. While high-throughput methods have made transcriptomics and proteomics datasets widely accessible and relatively economical to generate, systematic measurements of both mRNA and protein abundances under a wide range of different conditions are still relatively rare. Here we present a detailed, genome-wide transcriptomics and proteomics dataset of E. coli grown under 34 different conditions. Additionally, we provide measurements of doubling times and in-vivo metabolic fluxes through the central carbon metabolism. We manipulate concentrations of sodium and magnesium in the growth media, and we consider four different carbon sources glucose, gluconate, lactate, and glycerol. Moreover, samples are taken both in exponential and stationary phase, and we include two extensive time-courses, with multiple samples taken between 3â€‰hours and 2 weeks. We find that exponential-phase samples systematically differ from stationary-phase samples, in particular at the level of mRNA. Regulatory responses to different carbon sources or salt stresses are more moderate, but we find numerous differentially expressed genes for growth on gluconate and under salt and magnesium stress. Our data set provides a rich resource for future computational modeling of E. coli gene regulation, transcription, and translation.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead
