---
citation-template: v0.2.0
creation-date: 2025:01:20-20:23:22
---

%% Note Body --------------------------------------------------- %%
# Regulation of gene expression in flux balance models of metabolism

### Meta
- ** citekey **: covertRegulationGeneExpression2001
- ** authors **: [[Markus W. Covert|M. W. Covert]], C. H. Schilling, [[Bernhard Ã˜ Palsson]]
- ** year **: [[2001]]
- ** doi **: https://doi.org/10.1006/jtbi.2001.2405
- ** publication **: Journal of Theoretical Biology
- ** Web ** : [Open online]()


### Abstract:
Genome-scale metabolic networks can now be reconstructed based on annotated genomic data augmented with biochemical and physiological information about the organism. Mathematical analysis can be performed to assess the capabilities of these reconstructed networks. The constraints-based framework, with flux balance analysis (FBA), has been used successfully to predict time course of growth and by-product secretion, effects of mutation and knock-outs, and gene expression profiles. However, FBA leads to incorrect predictions in situations where regulatory effects are a dominant influence on the behavior of the organism. Thus, there is a need to include regulatory events within FBA to broaden its scope and predictive capabilities. Here we represent transcriptional regulatory events as time-dependent constraints on the capabilities of a reconstructed metabolic network to further constrain the space of possible network functions. Using a simplified metabolic/regulatory network, growth is simulated under various conditions to illustrate systemic effects such as catabolite repression, the aerobic/anaerobic diauxic shift and amino acid biosynthesis pathway repression. The incorporation of transcriptional regulatory events in FBA enables us to interpret, analyse and predict the effects of transcriptional regulation on cellular metabolism at the systemic level.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead