---
citation-template: v0.2.0
creation-date: 2025:02:27-20:25:32
---

%% Note Body --------------------------------------------------- %%
# Toward modeling metabolic state from single-cell transcriptomics

### Meta
- ** citekey **: hrovatinModelingMetabolicState2022
- ** authors **: Karin Hrovatin, David S. Fischer, Fabian J. Theis
- ** year **: [[2022]]
- ** doi **: https://doi.org/10.1016/j.molmet.2021.101396
- ** publication **: Molecular Metabolism
- ** Web ** : [Open online](https://www.sciencedirect.com/science/article/pii/S2212877821002532)


### Abstract:
Background Single-cell metabolic studies bring new insights into cellular function, which can often not be captured on other omics layers. Metabolic information has wide applicability, such as for the study of cellular heterogeneity or for the understanding of drug mechanisms and biomarker development. However, metabolic measurements on single-cell level are limited by insufficient scalability and sensitivity, as well as resource intensiveness, and are currently not possible in parallel with measuring transcript state, commonly used to identify cell types. Nevertheless, because omics layers are strongly intertwined, it is possible to make metabolic predictions based on measured data of more easily measurable omics layers together with prior metabolic network knowledge. Scope of Review We summarize the current state of single-cell metabolic measurement and modeling approaches, motivating the use of computational techniques. We review three main classes of computational methods used for prediction of single-cell metabolism: pathway-level analysis, constraint-based modeling, and kinetic modeling. We describe the unique challenges arising when transitioning from bulk to single-cell modeling. Finally, we propose potential model extensions and computational methods that could be leveraged to achieve these goals. Major Conclusions Single-cell metabolic modeling is a rising field that provides a new perspective for understanding cellular functions. The presented modeling approaches vary in terms of input requirements and assumptions, scalability, modeled metabolic layers, and newly gained insights. We believe that the use of prior metabolic knowledge will lead to more robust predictions and will pave the way for mechanistic and interpretable machine-learning models.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#bibnote
#Vault/MetXVault 