---
citation-template: v0.2.0
creation-date: #!cdate
---

%% Note Body --------------------------------------------------- %%
# Use of Adaptive Laboratory Evolution To Discover Key Mutations Enabling Rapid Growth of Escherichia coli K-12 MG1655 on Glucose Minimal Medium

### Meta
- ** citekey **: lacroixUseAdaptiveLaboratory2015
- ** authors **: Ryan A. LaCroix, Troy E. Sandberg, Edward J. O'Brien, Jose Utrilla, Ali Ebrahim, Gabriela I. Guzman, Richard Szubin, Bernhard O. Palsson, Adam M. Feist
- ** year **: [[2015]]
- ** doi **: https://doi.org/10.1128/AEM.02246-14
- ** publication **: Applied and Environmental Microbiology
- ** Web ** : [Open online](https://journals.asm.org/doi/10.1128/AEM.02246-14)


### Abstract:
ABSTRACT Adaptive laboratory evolution (ALE) has emerged as an effective tool for scientific discovery and addressing biotechnological needs. Much of ALE's utility is derived from reproducibly obtained fitness increases. Identifying causal genetic changes and their combinatorial effects is challenging and time-consuming. Understanding how these genetic changes enable increased fitness can be difficult. A series of approaches that address these challenges was developed and demonstrated using Escherichia coli K-12 MG1655 on glucose minimal media at 37Â°C. By keeping E. coli in constant substrate excess and exponential growth, fitness increases up to 1.6-fold were obtained compared to the wild type. These increases are comparable to previously reported maximum growth rates in similar conditions but were obtained over a shorter time frame. Across the eight replicate ALE experiments performed, causal mutations were identified using three approaches: identifying mutations in the same gene/region across replicate experiments, sequencing strains before and after computationally determined fitness jumps, and allelic replacement coupled with targeted ALE of reconstructed strains. Three genetic regions were most often mutated: the global transcription gene rpoB , an 82-bp deletion between the metabolic pyrE gene and rph , and an IS element between the DNA structural gene hns and tdk . Model-derived classification of gene expression revealed a number of processes important for increased growth that were missed using a gene classification system alone. The methods described here represent a powerful combination of technologies to increase the speed and efficiency of ALE studies. The identified mutations can be examined as genetic parts for increasing growth rate in a desired strain and for understanding rapid growth phenotypes.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead
