---
citation-template: v0.2.0
creation-date: 2025:02:12-10:15:41
---

%% Note Body --------------------------------------------------- %%
# Omic data from evolved E. Coli are consistent with computed optimal growth from genome-scale models

### Meta
- ** citekey **: lewisOmicDataEvolved2010
- ** authors **: Nathan E Lewis, Kim K Hixson, Tom M Conrad, Joshua A Lerman, Pep Charusanti, Ashoka D Polpitiya, Joshua N Adkins, Gunnar Schramm, Samuel O Purvine, Daniel Lopez-Ferrer, Karl K Weitz, Roland Eils, Rainer König, Richard D Smith, Bernhard Ø Palsson
- ** year **: [[2010]]
- ** doi **: https://doi.org/10.1038/msb.2010.47
- ** publication **: Molecular Systems Biology
- ** Web ** : [Open online](https://www.embopress.org/doi/full/10.1038/msb.2010.47)


### Abstract:
After hundreds of generations of adaptive evolution at exponential growth, Escherichia coli grows as predicted using flux balance analysis (FBA) on genome-scale metabolic models (GEMs). However, it is not known whether the predicted pathway usage in FBA solutions is consistent with gene and protein expression in the wild-type and evolved strains. Here, we report that >98% of active reactions from FBA optimal growth solutions are supported by transcriptomic and proteomic data. Moreover, when E. coli adapts to growth rate selective pressure, the evolved strains upregulate genes within the optimal growth predictions, and downregulate genes outside of the optimal growth solutions. In addition, bottlenecks from dosage limitations of computationally predicted essential genes are overcome in the evolved strains. We also identify regulatory processes that may contribute to the development of the optimal growth phenotype in the evolved strains, such as the downregulation of known regulons and stringent response suppression. Thus, differential gene and protein expression from wild-type and adaptively evolved strains supports observed growth phenotype changes, and is consistent with GEM-computed optimal growth states.

___

## View

%% Write here %%


## Results

> #Figure 2 In adaptive evolution through serial passaging, E. coli evolves to a higher growth rate and biomass yield at exponential growth. Growth rates and substrate uptake rates were acquired for each strain before and after adaptive evolution, as reported earlier ([[@ssParallelAdaptiveEvolution2005|Fong et al, 2005]]; Charusanti et al, submitted for publication). For growth on (A) glycerol, (B) lactate, and for (C) the Dpgi strain grown on glucose, the evolved strains (red) all improved their growth rate and biomass yield at exponential growth, compared with the unevolved parent strains (blue).
> ![[Pasted image 20250218113853.png]]

## Reference

- [Fong et al, 2005]  [[@ssParallelAdaptiveEvolution2005]] 
[1] https://doi.org/10.1371/journal.pone.0004809 missing
[2] https://doi.org/10.1128/JB.01976-07 missing
[3] https://doi.org/10.1038/nature08480 missing
[4] https://doi.org/10.1371/journal.pcbi.1000082 missing
[5] https://doi.org/10.1073/pnas.0609845104 missing
[6] https://doi.org/10.1073/pnas.0807227105 missing
[7] https://doi.org/10.1038/nbt.1582 missing
[8] https://doi.org/10.1186/gb-2009-10-10-r118 missing
[9] https://doi.org/10.1371/journal.pgen.0040035 missing
[10] https://doi.org/10.1073/pnas.120163297 missing
[11] https://doi.org/10.1038/nature03842 missing
[12] https://doi.org/10.1016/1044-0305(94)80016-2 missing
[13] https://doi.org/10.1038/msb4100155 missing
[14] https://doi.org/10.1038/nbt1401 missing
[15] https://doi.org/10.1016/j.ymben.2009.10.003 missing
[16] https://doi.org/10.1073/pnas.96.17.9721 missing
[17] https://doi.org/10.1146/annurev.biophys.32.110601.141854 missing
[18] https://doi.org/10.1101/gr.3832305 missing
[19] https://doi.org/10.1128/JB.185.21.6400-6408.2003 missing
[20] https://doi.org/10.1074/jbc.M510016200 missing
[21] https://doi.org/10.1038/ng1432 missing
[22] https://doi.org/10.1093/nar/gkm994 missing
[23] https://doi.org/10.1038/ng1906 missing
[24] https://doi.org/10.1128/jb.134.1.157-166.1978 missing
[25] https://doi.org/10.1128/AEM.00527-07 missing
[26] https://doi.org/10.1038/nature01149 missing
[27] https://doi.org/10.1371/journal.pgen.1000713 missing
[28] https://doi.org/10.1016/j.cell.2009.12.001 missing
[29] https://doi.org/10.1046/j.1365-2958.1996.5901313.x missing
[30] https://doi.org/10.1534/genetics.107.082040 missing
[31] https://doi.org/10.1073/pnas.91.15.6808 missing
[32] https://doi.org/10.1128/JB.00034-09 missing
[33] https://doi.org/10.1007/978-0-387-30440-3_329 missing
[34] https://doi.org/10.1016/j.ymben.2003.09.002 missing
[35] https://doi.org/10.1021/ac049043e missing
[36] https://doi.org/10.1038/msb.2009.82 missing
[37] https://doi.org/10.1093/bioinformatics/btn217 missing
[38] https://doi.org/10.1021/pr0498638 missing
[39] https://doi.org/10.1101/gr.2546004 missing
[40] https://doi.org/10.1093/bioinformatics/btq113 missing
[41] https://doi.org/10.1038/msb4100162 missing
[42] https://doi.org/10.1016/j.jtbi.2007.12.008 missing
[43] https://doi.org/10.1073/pnas.232349399 missing
[44] https://doi.org/10.1073/pnas.0406346102 missing
[45] https://doi.org/10.1073/pnas.1530509100 missing
[46] https://doi.org/10.1021/pr049965y missing
[47] https://doi.org/10.1126/science.278.5338.631 missing
[48] https://doi.org/10.1371/journal.pcbi.1000410 missing
[49] https://doi.org/10.1074/jbc.M606263200 missing
[50] https://doi.org/10.1371/journal.pcbi.1000312 missing
[51] https://doi.org/10.1111/j.1365-2958.2008.06229.x missing
[52] https://doi.org/10.1006/jtbi.1993.1203 missing
[53] https://doi.org/10.1128/jb.176.10.2807-2813.1994 missing
[54] https://doi.org/10.1002/mas.20071 missing


___
%% Tags  ------------------------------------------------------- %%
#review/ToRead
#Evolution 
#Data/Yield
#Data/Transcriptomic 
#EColi 
#PaperSON/Included