---
citation-template: v0.2.0
creation-date: 2024:11:11-12:13:47
---

%% Note Body --------------------------------------------------- %%
# Genome-scale metabolic network reconstructions of diverse <i>Escherichia</i> strains reveal strain-specific adaptations

### Meta
- ** citekey **: monkGenomescaleMetabolicNetwork2022
- ** authors **: Jonathan M. Monk
- ** year **: [[2022]]
- ** doi **: https://doi.org/10.1098/rstb.2021.0236
- ** publication **: Philosophical Transactions of the Royal Society B: Biological Sciences
- ** Web ** : [Open on line](https://royalsocietypublishing.org/doi/10.1098/rstb.2021.0236)


### Abstract:
Bottom-up approaches to systems biology rely on constructing a mechanistic basis for the biochemical and genetic processes that underlie cellular functions. Genome-scale network reconstructions of metabolism are built from all known metabolic reactions and metabolic genes in a target organism. A network reconstruction can be converted into a mathematical format and thus lend itself to mathematical analysis. Genome-scale models (GEMs) of metabolism enable a systems approach to characterize the pan and core metabolic capabilities of the Escherichia genus. In this work, GEMs were constructed for 222 representative strains of Escherichia across HC1100 levels spanning the known Escherichia phylogeny. The models were used to study Escherichia metabolic diversity and speciation on a large scale. The results show that unique strain-specific metabolic capabilities correspond to different species and nutrient niches. This work is a first step towards a curated reconstruction of pan- Escherichia metabolism. This article is part of a discussion meeting issue ‘Genomic population structures of microbial pathogens’.

___

## View

[[M9 minimal media]] was simulated by setting a lower bound of −1000 (allowing unlimited uptake) on the exchange reactions for $Ca^{2+}$, $Cl^{−}$, $CO^{2}$, $Co^{2+}$, $Cu^{2+}$, $Fe^{2+}$, $Fe^{3+}$, $H^{+}$, $H_{2}O$, $K^{+}$, $Mg^{2+}$, $Mn^{2+}$, $MoO^{2−}_4$, $Na^{+}$, $Ni^{2+}$, $SeO^{2−}_4$, $SeO^{2}_{3}$ and $Zn^{2+}$. A lower bound of −0.01 was placed on the cob(I)alamin exchange reaction. The default carbon source was glucose with a lower bound of −20, the default nitrogen source was $NH_4^{−}$ with a lower bound of −1000, the default phosphorus source was $HPO^2_4$ with a default bound of −1000 and the default sulfur source was $SO^{2−}_4$ with a default bound of −1000.





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead
