---
citation-template: v0.2.0
creation-date: 2025:01:23-17:18:16
---

%% Note Body --------------------------------------------------- %%
# Genome-scale models of metabolism and gene expression extend and refine growth phenotype prediction

### Meta
- ** citekey **: obrienGenomescaleModelsMetabolism2013
- ** authors **: Edward J. O'Brien, Joshua A. Lerman, Roger L. Chang, Daniel R. Hyduke, [[Bernhard Ø Palsson]]
- ** year **: [[2013]]
- ** doi **: https://doi.org/10.1038/msb.2013.52
- ** publication **: Molecular Systems Biology
- ** Web ** : [Open online]()


### Abstract:
Growth is a fundamental process of life. Growth requirements are well-characterized experimentally for many microbes; however, we lack a unified model for cellular growth. Such a model must be predictive of events at the molecular scale and capable of explaining the high-level behavior of the cell as a whole. Here, we construct an ME-Model for Escherichia coli–a genome-scale model that seamlessly integrates metabolic and gene product expression pathways. The model computes ~80% of the functional proteome (by mass), which is used by the cell to support growth under a given condition. Metabolism and gene expression are interdependent processes that affect and constrain each other. We formalize these constraints and apply the principle of growth optimization to enable the accurate prediction of multi-scale phenotypes, ranging from coarse-grained (growth rate, nutrient uptake, by-product secretion) to fine-grained (metabolic fluxes, gene expression levels). Our results unify many existing principles developed to describe bacterial growth.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead