---
citation-template: v0.2.0
creation-date: 2025:02:12-10:46:54
---

%% Note Body --------------------------------------------------- %%
# Cellular responses to reactive oxygen species are predicted from molecular mechanisms

### Meta
- ** citekey **: yangCellularResponsesReactive2019
- ** authors **: Laurence Yang, Nathan Mih, Amitesh Anand, Joon Ho Park, Justin Tan, James T. Yurkovich, Jonathan M. Monk, Colton J. Lloyd, Troy E. Sandberg, Sang Woo Seo, Donghyuk Kim, Anand V. Sastry, Patrick Phaneuf, Ye Gao, Jared T. Broddrick, Ke Chen, David Heckmann, Richard Szubin, Ying Hefner, Adam M. Feist, Bernhard O. Palsson
- ** year **: [[2019]]
- ** doi **: https://doi.org/10.1073/pnas.1905039116
- ** publication **: Proceedings of the National Academy of Sciences
- ** Web ** : [Open online](https://pnas.org/doi/full/10.1073/pnas.1905039116)


### Abstract:
Catalysis using iron–sulfur clusters and transition metals can be traced back to the last universal common ancestor. The damage to metalloproteins caused by reactive oxygen species (ROS) can prevent cell growth and survival when unmanaged, thus eliciting an essential stress response that is universal and fundamental in biology. Here we develop a computable multiscale description of the ROS stress response in Escherichia coli , called OxidizeME. We use OxidizeME to explain four key responses to oxidative stress: 1) ROS-induced auxotrophy for branched-chain, aromatic, and sulfurous amino acids; 2) nutrient-dependent sensitivity of growth rate to ROS; 3) ROS-specific differential gene expression separate from global growth-associated differential expression; and 4) coordinated expression of iron–sulfur cluster (ISC) and sulfur assimilation (SUF) systems for iron–sulfur cluster biosynthesis. These results show that we can now develop fundamental and quantitative genotype–phenotype relationships for stress responses on a genome-wide basis.

___

## View

%% Write here %%





___
%% Tags  ------------------------------------------------------- %%
#review/ToRead