https://doi.org/10.1038/s41564-019-0610-7

[1] Monod, J. Recherches sur la Croissance des Cultures Bactériennes (Hermann & Cie., 1942).
[2] (1947) Monod, J. The phenomenon of enzymatic adaptation—and its bearings on problems of genetics and cellular differentiation. Growth 11, 223–289 (1947).
[3] https://doi.org/10.1515/9783110879476 Müller-Hill, B. The lac Operon: a Short History of a Genetic Paradigm (Walter de Gruyter, 1996).
[4] https://doi.org/10.1128/MMBR.00024-06 (2006) Deutscher, J., Francke, C. & Postma, P. W. How phosphotransferase system-related protein phosphorylation regulates carbohydrate metabolism in bacteria. Microbiol. Mol. Biol. Rev. 70, 939–1031 (2006).
[5] https://doi.org/10.1016/j.jtbi.2006.08.007 (2007) Narang, A. & Pilyugin, S. S. Bacterial gene regulation in diauxic and non-diauxic growth. J. Theor. Biol. 244, 326–348 (2007).
[6] https://doi.org/10.1128/jb.93.4.1397-1401.1967 (1967) Loomis, W. F. & Magasanik, B. Glucose-lactose diauxie in Escherichia coli. J. Bacteriol. 93, 1397–1401 (1967).
[7] https://doi.org/10.1046/j.1365-2443.1996.24025.x (1996) Inada, T., Kimata, K. & Aiba, H. Mechanism responsible for glucose-lactose diauxie in Escherichia coli: challenge to the cAMP model. Genes Cells 1, 293–301 (1996).
[8] https://doi.org/10.1128/aem.62.5.1493-1499.1996 (1996) Lendenmann, U., Snozzi, M. & Egli, T. Kinetics of the simultaneous utilization of sugar mixtures by Escherichia coli in continuous culture. Appl. Environ. Microbiol. 62, 1493–1499 (1996).
[9] https://doi.org/10.1002/bit.260090206 (1967) Baidya, T. K. N., Webb, F. C. & Lilly, M. D. The utilization of mixed sugars in continuous fermentation. I. Biotechnol. Bioeng. 9, 195–204 (1967).
[10] https://doi.org/10.1002/bit.260090207 (1967) Harte, M. J. & Webb, F. C. Utilisation of mixed sugars in continuous fermentation. II. Biotechnol. Bioeng. 9, 205–221 (1967).
[11] https://doi.org/10.1098/rstb.1982.0055 (1982) Harder, W. & Dijkhuizen, L. Strategies of mixed substrate utilization in microorganisms. Philos. Trans. R. Soc. Lond. B Biol. Sci. 297, 459–480 (1982).
[12] https://doi.org/10.1111/j.1574-6968.1990.tb04084.x (1990) Wanner, U. & Egli, T. Dynamics of microbial growth and cell composition in batch culture. FEMS Microbiol. Rev. 6, 19–43 (1990).
[13] https://doi.org/10.1007/BF00871224 (1993) Egli, T., Lendenmann, U. & Snozzi, M. Kinetics of microbial growth with mixtures of carbon sources. Antonie Van Leeuwenhoek 63, 289–298 (1993).
[14] https://doi.org/10.15252/msb.20145537 (2015) Hermsen, R., Okano, H., You, C., Werner, N. & Hwa, T. A growth-rate composition formula for the growth of E. coli on co-utilized carbon substrates. Mol. Syst. Biol. 11, 801 (2015).
[15] https://doi.org/10.1146/annurev.mi.30.100176.002535 (1976) Lin, E. C. Glycerol dissimilation and its regulation in bacteria. Annu. Rev. Microbiol. 30, 535–578 (1976).
[16] https://doi.org/10.1038/nature12446 (2013) You, C. et al. Coordination of bacterial proteome with metabolism by cyclic AMP signalling. Nature 500, 301–306 (2013).
[17] https://doi.org/10.1016/S0021-9258(18)93859-7 (1964) Koch, J. P., Hayashi, S. & Lin, E. C. The control of dissimilation of glycerol and l-α-glycerophosphate in Escherichia coli. J. Biol. Chem. 239, 3106–3108 (1964).
[18] https://doi.org/10.1016/S0021-9258(18)42670-1 (1992) Weissenborn, D. L., Wittekindt, N. & Larson, T. J. Structure and regulation of the glpFK operon encoding glycerol diffusion facilitator and glycerol kinase of Escherichia coli K-12. J. Biol. Chem. 267, 6122–6131 (1992).
[19] https://doi.org/10.1126/science.153.3737.755 (1966) Zwaig, N. & Lin, E. C. Feedback inhibition of glycerol kinase, a catabolic enzyme in Escherichia coli. Science 153, 755–757 (1966).
[20] https://doi.org/10.1128/JB.183.11.3336-3344.2001 (2001) Holtman, C. K., Pawlyk, A. C., Meadow, N. D. & Pettigrew, D. W. Reverse genetics of Escherichia coli glycerol kinase allosteric regulation and glucose control of glycerol utilization in vivo. J. Bacteriol. 183, 3336–3344 (2001).
[21] https://doi.org/10.1073/pnas.1202582110 (2013) Kochanowski, K. et al. Functioning of a metabolic flux sensor in Escherichia coli. Proc. Natl Acad. Sci. USA 110, 1130–1135 (2013).
[22] https://doi.org/10.15252/msb.20145697 (2015) Hui, S. et al. Quantitative proteomic analysis reveals a simple strategy of global resource allocation in bacteria. Mol. Syst. Biol. 11, 784 (2015).
[23] https://doi.org/10.1128/jb.178.10.2846-2852.1996 (1996) Pettigrew, D. W., Liu, W. Z., Holmes, C., Meadow, N. D. & Roseman, S. A single amino acid change in Escherichia coli glycerol kinase abolishes glucose control of glycerol utilization in vivo. J. Bacteriol. 178, 2846–2852 (1996).
[24] https://doi.org/10.15252/msb.20167402 (2017) Kochanowski, K. et al. Few regulatory metabolites coordinate expression of central metabolic genes in Escherichia coli. Mol. Syst. Biol. 13, 903 (2017).
[25] https://doi.org/10.1074/jbc.M110.195305 (2011) Applebee, M. K., Joyce, A. R., Conrad, T. M., Pettigrew, D. W. & Palsson, B. Functional and metabolic effects of adaptive glycerol kinase (GLPK) mutants in Escherichia coli. J. Biol. Chem. 286, 23150–23159 (2011).
[26] https://doi.org/10.1128/JB.00819-07 (2007) Bettenbrock, K. et al. Correlation between growth rates, EIIACrr phosphorylation, and intracellular cyclic AMP levels in Escherichia coli K-12. J. Bacteriol. 189, 6891–6900 (2007).
[27] https://doi.org/10.1038/nature24299 (2017) Erickson, D. W. et al. A global resource allocation strategy governs growth transition kinetics of Escherichia coli. Nature 551, 119–123 (2017).
[28] https://doi.org/10.1093/nar/gkw1003 (2017) Keseler, I. M. et al. The EcoCyc database: reflecting new knowledge about Escherichia coli K-12. Nucleic Acids Res. 45, D543–D550 (2017).
[29] https://doi.org/10.1042/bj0820489 (1962) Mandelstam, J. The repression of constitutive beta-galactosidase in Escherichia coli by glucose and other carbon sources. Biochem. J. 82, 489–493 (1962).
[30] https://doi.org/10.1016/j.jtbi.2013.11.015 (2014) Müller, S., Regensburger, G. & Steuer, R. Enzyme allocation problems in kinetic metabolic networks: optimal solutions are elementary flux modes. J. Theor. Biol. 347, 182–190 (2014).
[31] https://doi.org/10.1111/febs.12722 (2014) Wortel, M. T., Peters, H., Hulshof, J., Teusink, B. & Bruggeman, F. J. Metabolic states with maximal specific rate carry flux through an elementary flux mode. FEBS J. 281, 1547–1555 (2014).
[32] https://doi.org/10.1038/s41467-019-09261-3 (2019) Wang, X., Xia, K., Yang, X. & Tang, C. Growth strategy of microbes on mixed carbon sources. Nat. Commun. 10, 1279 (2019).
[33] https://doi.org/10.1007/s10295-008-0327-8 (2008) Kumar, R., Singh, S. & Singh, O. V. Bioconversion of lignocellulosic biomass: biochemical and molecular perspectives. J. Ind. Microbiol. Biotechnol. 35, 377–391 (2008).
[34] https://doi.org/10.1007/s00253-010-2839-1 (2010) Kim, J. H., Block, D. E. & Mills, D. A. Simultaneous consumption of pentose and hexose sugars: an optimal microbial phenotype for efficient fermentation of lignocellulosic biomass. Appl. Microbiol. Biotechnol. 88, 1077–1085 (2010).
[35] https://doi.org/10.5483/BMBRep.2012.45.2.59 (2012) Vinuselvi, P., Kim, M. K., Lee, S. K. & Ghim, C. M. Rewiring carbon catabolite repression for microbial cell factory. BMB Rep. 45, 59–70 (2012).
[36] https://doi.org/10.1128/JB.185.18.5611-5626.2003 (2003) Soupene, E. et al. Physiological studies of Escherichia coli strain MG1655: growth defects and apparent cross-regulation of gene expression. J. Bacteriol. 185, 5611–5626 (2003).
[37] https://doi.org/10.1128/jb.176.20.6324-6333.1994 (1994) Csonka, L. N., Ikeda, T. P., Fletcher, S. A. & Kustu, S. The accumulation of glutamate is necessary for optimal growth of Salmonella typhimurium in media of high osmolality but not induction of the proU operon. J. Bacteriol. 176, 6324–6333 (1994).
[38] https://doi.org/10.15252/msb.20156178 (2015) Basan, M. et al. Inflating bacterial cells by increased protein synthesis. Mol. Syst. Biol. 11, 836 (2015).
[39] https://doi.org/10.1128/jb.112.1.183-187.1972 (1972) Morrissey, A. T. & Fraenkel, D. G. Suppressor of phosphofructokinase mutations of Escherichia coli. J. Bacteriol. 112, 183–187 (1972).
[40] https://doi.org/10.1002/j.1460-2075.1991.tb08056.x (1991) de Lorenzo, V., Herrero, M., Metzke, M. & Timmis, K. N. An upstream XylR- and IHF-induced nucleoprotein complex regulates the sigma 54-dependent Pu promoter of TOL plasmid. EMBO J. 10, 1159–1167 (1991).
[41] https://doi.org/10.1016/j.cell.2009.12.001 (2009) Klumpp, S., Zhang, Z. & Hwa, T. Growth rate-dependent global effects on gene expression in bacteria. Cell 139, 1366–1375 (2009).
[42] https://doi.org/10.1073/pnas.120163297 (2000) Datsenko, K. A. & Wanner, B. L. One-step inactivation of chromosomal genes in Escherichia coli K-12 using PCR products. Proc. Natl Acad. Sci. USA 97, 6640–6645 (2000).
[43] https://doi.org/10.1038/msb.2012.46 (2012) Kim, M. et al. Need-based activation of ammonium uptake in Escherichia coli. Mol. Syst. Biol. 8, 616 (2012).
[44] https://doi.org/10.1093/clinchem/29.3.538 (1983) McGowan, M. W., Artiss, J. D., Strandbergh, D. R. & Zak, B. A peroxidase-coupled method for the colorimetric determination of serum triglycerides. Clin. Chem. 29, 538–542 (1983).
[45] https://doi.org/10.1038/nchembio.1437 (2014) Sévin, D. C. & Sauer, U. Ubiquinone accumulation improves osmotic-stress tolerance in Escherichia coli. Nat. Chem. Biol. 10, 266–272 (2014).
[46] https://doi.org/10.1093/nar/27.1.29 (1999) Ogata, H. et al. KEGG: Kyoto Encyclopedia of Genes and Genomes. Nucleic Acids Res. 27, 29–34 (1999).
[47] https://doi.org/10.1038/nmeth.2019 (2012) Schindelin, J. et al. Fiji: an open-source platform for biological-image analysis. Nat. Methods 9, 676–682 (2012).
[48] https://doi.org/10.1016/S0021-9258(18)57021-6 (1949) Gornall, A. G., Bardawill, C. J. & David, M. M. Determination of serum proteins by means of the biuret reaction. J. Biol. Chem. 177, 751–766 (1949).
[49] https://doi.org/10.1016/S0021-9258(17)32989-7 (1976) Saier, M. H. & Roseman, S. Sugar transport. Inducer exclusion and regulation of the melibiose, maltose, glycerol, and lactose transport systems by the phosphoenolpyruvate:sugar phosphotransferase system. J. Biol. Chem. 251, 6606–6615 (1976).
[50] https://doi.org/10.1111/j.1432-1033.1977.tb11906.x (1977) Sandermann, H. Jr. β-d-Galactoside transport in Escherichia coli: substrate recognition. Eur. J. Biochem. 80, 507–515 (1977).
[51] https://doi.org/10.1016/S0021-9258(19)45412-4 (1982) Stock, J. B., Waygood, E. B., Meadow, N. D., Postma, P. W. & Roseman, S. Sugar transport by the bacterial phosphotransferase system. The glucose receptors of the Salmonella typhimurium phosphotransferase system. J. Biol. Chem. 257, 14543–14552 (1982).
[52] https://doi.org/10.1021/bi00295a019 (1983) Misset, O., Blaauw, M., Postma, P. W. & Robillard, G. T. Bacterial phosphoenolpyruvate-dependent phosphotransferase system. Mechanism of the transmembrane sugar translocation and phosphorylation. Biochemistry 22, 6163–6170 (1983).
