---
creation-date: "2023:01:12-11:03:19"
basic-note-template: "v0.2.0"
---

%% Note Body --------------------------------------------------- %%

%% Write here %%


#### [[@herrgardIntegratedAnalysisRegulatory2006]]

> ### Prediction of growth phenotypes
> In order to provide data for an evaluation of the predictive power of the integrated regulatory and metabolic network model iMH805/837, we performed quantitative growth phenotyping experiments on 12 different carbon sources with 10 TF deletion strains. #Project/PaperSON 

#### [[@covertIntegratingHighthroughputComputational2004]]

#### [[@covertRegulationGeneExpression2001]]

#### [[@wintermuteEmergentCooperationMicrobial2010]]

#### [[@forsterLargeScaleEvaluationSilico2003]]

> - Large-scale evaluation of in silico gene deletions in Saccharomyces cerevisiae

A large-scale in silico evaluation of gene deletions in Saccharomyces cerevisiae was conducted using a genome-scale reconstructed metabolic model. The effect of 599 single gene deletions on cell viability was simulated in silico and compared to published experimental results.

#### [[@maserAvoidingAminoAcid2019]]

#EColi #ComplexMedium #SupplemmentaryMaterials
#NutrientLimited 

> #Figure 5. Measured amino acid consumption compared to the amino acid requirements in the biomass-based medium experiments. Sampling points, where aspartate and threonine were still present, were used for calculations. Serine (consumption was >2 mmol gDCW1 and is not shown in the figure), cysteine, aspartate, threonine and arginine were consumed more than the presumed need for protein and nucleotide synthesis, while asparagine, alanine and glycine were underutilize
> ![[Pasted image 20241205084427.png|400]]

> #Table 1. [[E. coli]] [[BW25113]] growth characteristics in the media used in this work In acid-hydrolysed casein, biomass-based and arginine addition medium overall growth and the division into two different phases (exponential growth phase and changing growth phase) is given. Yield calculation is shown in C-mole terms and considers carbon coming from both glucose and amino 
> ![[Pasted image 20241205085046.png|600]]

#### https://github.com/yinonbaron/ecoli_composition/blob/master/rna_protein_ratio.ipynb
- #TOCHECK 
- ![[Pasted image 20241016190118.png|400]]

#### [[@wuEnzymeExpressionKinetics2023]]

#### [[@monkGenomescaleMetabolicReconstructions2013]]

#### [[@liuMinimizationProteomeReallocation2024]]

- #CHECKED

#### [[@tweeddaleEffectSlowGrowth1998]]

- Clear trends were apparent in the pool sizes of several of the 70 most abundant metabolites extracted from bacteria growing in glucose-limited chemostats at different growth rates.

#### [[@monkGenomescaleMetabolicNetwork2022]]

#### [[@drosteThermodynamicsUnicellularLife2024]]

- Metabolic Shift fr yeast and E. coli

#### [[@priceComparisonCostsBenefits2016]]

#### [[@schmidtQuantitativeConditiondependentEscherichia2016]]

Schmidt et al. recently published a proteomics data Resource13 covering ~55% of the predicted E. coli genes (>95% by mass) under 22 experimental conditions.

#### [[@obrienQuantificationClassificationColi2016]]

#### [[@yangPrinciplesProteomeAllocation2016]]

![[Pasted image 20241029030628.png|300]]

#### [[@kleijnGrowthratedependentNutrientspecificGene2022]]

![[Pasted image 20241001073616.png|300]]

#### [[@calabreseProteinDegradationSets2022]]

![[Pasted image 20241001073103.png|300]]

#### [[@ghiaciHighlyParallelizedLaboratory2024]]

#### [[@monkMultiomicsQuantificationSpecies2016]]

![[Pasted image 20240929213628.png|300]]

#### [[@baldazziResourceAllocationAccounts2023]]
> 
> #Figure 
> ![[Pasted image 20240924100052.png|300]]

> #Figure ICTP lecture during "[Advanced School on Quantitative Principles in Microbial](https://www.youtube.com/playlist?list=PLRwcSE2bmyByPnSauvAJNKhbR87BbK8mB)"
> ![[Pasted image 20240929223035.png|300]]

#### [[@gerosaPseudotransitionAnalysisIdentifies2015]]

-  Pseudo-transition Analysis Identifies the Key Regulators of Dynamic Metabolic Adaptations from Steady-State Data

> We next validated intracellular flux predictions for the Optimal and Generalist models using metabolic flux analysis (MFA) data by Gerosa et al.24 for 7 carbon sources: acetate, fructose, galactose, glucose, glycerol, pyruvate, and succinate.


> #Figure 
> ![[Pasted image 20240718120513.png]]

#### [[@brLargescale13CfluxAnalysis2011]]

#Figure ![[Pasted image 20240718115421.png|400]]

> #Figure #Question Why the change in flux? If is galactose is very similar to glucose... It is because the growth rate? 

> ![[Pasted image 20240719131733.png|400]]

#### [[@carnicerMacromolecularElementalComposition2009]]

#### [[@heckmannMachineLearningApplied2018]]

> In this work, independent component analysis (ICA) was applied to a compendium of existing fitness data from randomly barcoded transposon insertion sequencing (RB-TnSeq) of P. putida KT2440 grown in 179 unique experimental conditions.

#### [[@priceMutantPhenotypesThousands2018]]

> Fitness Browser: https://fit.genomics.lbl.gov/cgi-bin/myFrontPage.cgi 
> Here, to investigate the functions of these genes, we present genome-wide mutant fitness data from 32 diverse bacteria across dozens of growth conditions.

## E coli cultures

#### [[@vadiaGrowthRateCell2015a]]

Check References. It includes a lot of studies switching nutrient.

#### [[@wangGenomicTranscriptomicMetabolic2020]]

> Here, in this study, genomic, transcriptomic, and 13C-metabolic flux analyses (13C-MFA) of an evolved E. coli strain adapted to BCHA tolerance were conducted.

#### [[@lewisOmicDataEvolved2010]]

> Multiple strains of E. coli were subjected to adaptive evolution through serial passaging in three different M9-minimal media conditions: lactate, glycerol, and glucose (glucose grown strains had the glycolytic gene pgi deleted to perturb the normal flux into glycolysis). For each growth condition, three to six replicates of the adaptive process were performed in parallel until each strain had reached and maintained a steadygrowth rate, which typically took 700–1000 generations (see Fong et al, 2005, 2006 for details). Through adaptive evolution, all strains improved their growth rate and efficiency in converting substrate to biomass (yield) within the exponential growth phase (Figure 2).
> Fifty quantitative proteomic data sets were obtained from the wild-type and evolved strains. Within these data sets, 983 proteins were identified with high confidence, of which 731 were identified in all strains. An extended discussion on methods and an analysis of data content and quality can be found in the Supplementary information. Transcriptomic data for strains corresponding to two of the three growth conditions (lactate and glycerol) have been published earlier (Fong et al, 2005) and are also analyzed alongside the proteomic data in this study using the E. coli GEM as a context for the analysis.

#### [[@bhadra-loboAssessmentTranscriptomicConstraintbased2020]]

> In this paper, we formed a dataset of transcriptomic data coupled with corresponding 13C-MFA flux data for 21 experimental conditions in different unicellular organisms grown on varying carbon substrates and conditions. Three computational flux-balance analysis (FBA) methods were comparatively assessed. The results show when uptake rates of carbon sources and key metabolites are known, transcriptomic data provides no significant advantage over constraint-based metabolic modeling (average correlation coefficients, transcriptomic E-Flux2 0.725 and SPOT 0.650 vs non-transcriptomic pFBA 0.768). When uptake rates are unknown, however, predictions obtained utilizing transcriptomic data are generally good and significantly better than those obtained using constraint-based metabolic modeling alone (E-Flux2 0.385 and SPOT 0.583 vs pFBA 0.237). Thus, transcriptomic data coupled with constraint-based metabolic modeling is a promising method to obtain intracellular flux estimates in microorganisms, particularly in cases where uptake rates of key metabolites cannot be easily determined, such as for growth in complex media or _in vivo_ conditions.

> Data and model for _E_. _coli_: For _E_. _coli_, both the measured gene expression (single color microarray) and 13C flux data were obtained from a previous study by Gerosa et al. [[17](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7480874/#pone.0238689.ref017)]. In this study, data were measured from _E_. _coli_ wild type BW25113 cells growing exponentially on eight different carbon sources: glucose, galactose, gluconate, fructose, glycerol, pyruvate, acetate, and succinate. We used iJO1366 [[18](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7480874/#pone.0238689.ref018)] as the genome-scale metabolic model.


#### [[@kimEFlux2SPOTValidated2016]]

We examined _E_. _coli_ and _S_. _cerevisiae_ as representative prokaryotic and eukaryotic microorganisms respectively. The predictive accuracy of our algorithm was validated by calculating the uncentered Pearson correlation between predicted fluxes and measured fluxes. To this end, we compiled 20 experimental conditions (11 in _E_. _coli_ and 9 in _S_. _cerevisiae_), of transcriptome measurements coupled with corresponding central carbon metabolism intracellular flux measurements determined by 13C metabolic flux analysis (13C-MFA), which is the largest dataset assembled to date for the purpose of validating inference methods for predicting intracellular fluxes.

### Chemostat 

#TODO check/add all this paper's data

#### [[@moriConstrainedAllocationFlux2016]]

#### [[@jeniorTranscriptomeguidedParsimoniousFlux2020]]

> status: #TODO

> To accomplish this goal, we performed a comparative analysis using three separate transcriptomic sequencing datasets, each derived from well characterized in vitro media conditions including aerobic LB rich media, aerobic M9+glucose defined minimal media, and anaerobic M9+glucose defined minimal media. Each dataset was collected at approximately the same mid-exponential phase of growth [47,48], and contextualization of iJO1366 with RIPTiDe took only seconds to complete (S3B Fig).

#### [[@millardMetabolicRegulationSufficient2017]]
> status #TODO

> #Figure 4. Exploration of the solution space of E. coli metabolism with and without metabolic regulation. Density of scatter plots between glucose uptake and growth rates sampled using the stoichiometric model (A, without metabolic regulation) or the kinetic model (B, with metabolic regulation). Shades of white to blue denote null to high frequency, respectively. Red dots are measurements obtained from 254 independent cultivations of wild-type and mutant E. coli strains on glucose, carried out under a wide range of cultivation conditions.
> ![[Pasted image 20230913224731.png|600]]


#### [[@folsomPhysiologicalBiomassElemental2015]]
> status: #DONE

#### [[@kayserMetabolicFluxAnalysis2005]]
> status: #DONE

#### [[@ihssenSpecificGrowthRate2004]] 
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)

#### [[@huaAnalysisGeneExpression2004]]
	- Have differential expression between C and N limited cultures
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)

#### [[@huaResponsesCentralMetabolism2003]]
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)
	- C and N limited
	- wild and mutant
	- external  & internal fluxes 
	- Biomass composition

#### [[@emmerlingMetabolicFluxResponses2002]]
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)
	- Biomass composition
	- external  & internal fluxes 
	- C and N limited
	- wild and mutant

#### [[@segreAnalysisOptimalityNatural2002]]

#### [[@seniorRegulationNitrogenMetabolism1975]]
> status: #DONE
	
	 - (see folsomPhysiologicalBiomassElemental2015 supp 4)

#### [[@sukmariniMetabolicRegulationEscherichia2010]]
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)

#### [[@snoepDifferencesSensitivityNADH1993]]
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)

#### [[@sauerMetabolicFluxRatio1999]]
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)

#### [[@yangAnalysisEscherichiaColi2003]]
- abdel-hamidPyruvateOxidaseContributes2001
	- (see folsomPhysiologicalBiomassElemental2015 supp 4)

#### [[@vanthofHighqualityGenomescaleMetabolic2022]]
- @schultzPredictingInternalCell2015
- Synergy between (13)C-metabolic flux analysis and flux balance analysis for understanding metabolic adaptation to anaerobiosis in E. coli

#### [[@wintermuteObjectiveFunctionExploiting2013]]
#Figure **Flux correlations and prediction errors for the PSEUDO method. (A, B, C)** Flux predictions derived using the FBA, MOMA and PSEUDO objective functions are plotted against 320 flux measurements taken from 12 different mutants in the Tomita data set. Flux values are reported as a percentage of the glucose uptake rate in each mutant.
	![[Pasted image 20230311233640.png]]

> The Tomita data set includes 31 fluxes measured in chemostat cultures of 24 deletion mutants for enzymes in central carbon processing [25].

> The Sauer data set measures 24 central carbon fluxes in 91 transcription factor deletion mutants [37].
> 
#### [[@schuetzSystematicEvaluationObjective2007]]

#### [[@ishiiMultipleHighthroughputAnalyses2007]]
	 - see supp material (TABLE S5)

#### [[@nakahigashiSystematicPhenomeAnalysis2009]]

#### [[@brLargescale13CfluxAnalysis2011]]
> #Figure 1: Absolute metabolic fluxes in E. coli during aerobic growth on glucose (A) or galactose (B). Flux arrows are drawn in proportion to the substrate uptake rates for each condition. The numbers represent absolute flux values ($mmol gCDW^{-1} h^{-1}$). One of two replicate experiments is shown (Supplementary Tables 2 and 3). The presented fluxes are from one of two independent experiments and were obtained by 13C-constrained flux analysis using the software FiatFlux (Zamboni et al, 2005). They were independently confirmed by flux estimation with a whole isotopologue model (Kleijn et al, 2005; van Winden et al, 2005). Generally, the deviation between the two independent experiments was 1–5%, and whole isotopologue sensitivity analysis through addition of Gaussian noise confirmed accurate estimation for all major fluxes (Supplementary Tables 2 and 3).
> ![[Pasted image 20230312012353.png]]

### [[S. cerevisiae]]

#### [[@snitkinModeldrivenAnalysisExperimentally2008]]

> Results
In this study, we measured growth phenotypes of 465 _Saccharomyces cerevisiae_ gene deletion mutants under 16 metabolically relevant conditions and integrated them with the corresponding flux balance model predictions.

#### [[@gianchandaniPredictingBiologicalSystem2008]]

> A flux distribution for the central metabolic network was previously characterized experimentally [22, 23]. This distribution was obtained by GC-MS tracing of 13C-labeled isotopomers, and corresponds to yeast cells cultivated in batch culture with glucose as the limiting substrate and at a maximum specific growth rate, _μ_max, of 0.37 h-1. (See [23] for complete experimental details.)

#### [[@osterlundMappingConditiondependentRegulation2013a]]

> #Figure **Model validation by comparing** **_in silico_** **prediction of the specific growth rate with experimental data.** Growth phenotypes were collected from literature and compared to simulated values for chemostat cultivations at four different conditions, nitrogen limited aerobic (green) and anaerobic (red), carbon limited aerobic (blue) and anaerobic (white).
> ![[Pasted image 20230425133918.png|300]]

> The iTO977 model was used as a scaffold for integrated data analysis. Chemostat cultivations were performed at a dilution rate of 0.05 h-1 under the same four different conditions as used for simulations above (carbon limited, nitrogen limited, aerobic and anaerobic) and microarray data as well as external and intracellular flux measurements was obtained for each condition[20].

> The enzymes that have a direct link between change in expression and change in flux might be good targets for over-expression in order to direct the flux toward i.e. a desired product, since there is a coupling between expression and flux. We therefore used the random sampling algorithm[15] to identify reactions that change both in flux and transcription. These reactions are referred to as transcriptionally controlled reactions and can be seen as a subset of the global set of reactions that change transcriptionally.


#### [[@hazelwoodIdentityGrowthLimitingNutrient2009]]

#NutrientLimited #Chemostat

#### [[@daran-lapujadeChemostatBasedMicroArrayAnalysis2008]]

#NutrientLimited #Chemostat #Transcriptomic

#### [[@natarajanTranscriptionalProfilingShows2001]]

#NutrientLimited #Chemostat

#### [[@taiTwodimensionalTranscriptomeAnalysis2005]]

#NutrientLimited #Chemostat

#### [[@piperReproducibilityOligonucleotideMicroarray2002]]

#NutrientLimited #GlucoseLimited #Chemostat 

#### [[@daran-lapujadeRoleTranscriptionalRegulation2004]]

#NutrientLimited #Chemostat 

#### [[@boerGenomewideTranscriptionalResponses2003]]

#NutrientLimited #Chemostat 

#### [[@boerGrowthlimitingIntracellularMetabolites2010]]

#NutrientLimited #Chemostat 

> Saccharomyces cerevisiae was cultivated in chemostats using five different limiting nutrients, each at five different dilution rates (0.05, 0.11, 0.16, 0.22 and 0.30 1/h ).

#### [[@wuGlobalAnalysisNutrient2004]]

#NutrientLimited #Chemostat 

#### [[@saldanhaNutritionalHomeostasisBatch2004a]]

#NutrientLimited #Chemostat 

___

### CHO

#### [[@vergaraHighGlucoseLow2018]]
 > ![[Pasted image 20230318104122.png| 400]]

#### [[@szeliovaInclusionMaintenanceEnergy2021]]

> #Figure Experimental vs. predicted growth rates (A) and intracellular fluxes (B) after the addition of mATP as constraint. Results are shown for R_biomass_cho as the objective function. RE—relative error. The legend in panel (A) indicates the publication and the used CHO cell line (if the information was available). Empty symbols indicate non-producers.
> ![[Pasted image 20230407195938.png|500]]

#### [[@ahnMetabolicFluxAnalysis2011]]

> ![[Pasted image 20230401103804.png|300]]

#### [[@goudarMetabolicFluxAnalysis2010]]

> ![[Pasted image 20230401105111.png]]

#### [[@mcateepereira13FluxAnalysis2018]]

#### [[@dietmairMetaboliteProfilingCHO2012]]  

#Batch culture with different media. 

> Table I. Metabolite consumption and production rates of HyQ, SFM, and CDCHO cultures during exponential phase.
> ![[Pasted image 20230611162812.png|300]]


### Other Mammals

#### [[@follstadMetabolicFluxAnalysis1999]]

#GlutamineLimited #Chemostat

> #Table Measured metabolite production and consumption rates (×10^{−10} mmol/viable cell/h)^a
> 
> ![[Pasted image 20230611185929.png]]
>
> ^a These values were calculated taking into consideration the biosynthesis demand for amino acids and other metabolites. Negative values indicate metabolite consumption, whereas positive values indicate metabolite production.


#### [[@vriezenEffectsGlutamineSupply1997]]

#Chemostat #NutrientLimited #Glutamine


#### [[@rathCharacterisationCellGrowth2017]]
> status: #DONE

%% Tags ------------------------------------------------------- %%
#ProjectRoot 
___
